File: Strata/trunk/java/tests/StrataCommonTest/src/strata1/common/producerconsumer/CountRequestDisruptorDispatcher.java
// ##########################################################################
// # File Name:	CountRequestDisruptorRouter.java
// #
// # Copyright:	2014, Sapientia Systems, LLC. All Rights Reserved.
// #
// # License:	This file is part of the StrataCommonTest Framework.
// #
// #   			The StrataCommonTest Framework is free software: you 
// #			can redistribute it and/or modify it under the terms of 
// #			the GNU Lesser General Public License as published by
// #    		the Free Software Foundation, either version 3 of the 
// #			License, or (at your option) any later version.
// #
// #    		The StrataCommonTest Framework is distributed in the 
// #			hope that it will be useful, but WITHOUT ANY WARRANTY; 
// #			without even the implied warranty of MERCHANTABILITY or 
// #			FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser 
// #			General Public License for more details.
// #
// #    		You should have received a copy of the GNU Lesser 
// #			General Public License along with the StrataCommonTest
// #			Framework. If not, see http://www.gnu.org/licenses/.
// ##########################################################################

package strata1.common.producerconsumer;

/****************************************************************************
 * 
 * @author 		
 *     Sapientia Systems
 * @conventions	
 *     <a href="{@docRoot}/NamingConventions.html">Naming Conventions</a>
 */
public 
class CountRequestDisruptorDispatcher
    extends    DisruptorDispatcher<CountRequest>
    implements ICountRequestDispatcher
{

    /************************************************************************
     * Creates a new {@code CountRequestDisruptorRouter}. 
     *
     * @param factory
     * @param bufferSize
     */
    public 
    CountRequestDisruptorDispatcher(int bufferSize)
    {
        super( new CountRequestEventFactory(),bufferSize );
    }

}

// ##########################################################################

File: csharp/tests/Strata.FoundationTest/ProducerConsumer/CountRequestDisruptorDispatcher.cs
//  ##########################################################################
//  # File Name: CountRequestDisruptorDispatcher.cs
//  # Copyright: 2011-2018, Sapientia Systems, LLC.
//  ##########################################################################

using System;

namespace Strata.Foundation.ProducerConsumer
{
    //////////////////////////////////////////////////////////////////////////
    /// <summary>
    /// 
    /// </summary>
    ///  
    public
    class CountRequestDisruptorDispatcher:
        DisruptorDispatcher<CountRequest>,
        ICountRequestDispatcher
    {
        public
        CountRequestDisruptorDispatcher(int bufferSize): 
            base( Create,bufferSize ) {}

        public static Event<CountRequest>
        Create()
        {
            return new Event<CountRequest>();
        }
    }
}

//  ##########################################################################
